### DISCUSSION TITLE
Regression with custom interception

### DISCUSSION THREAD

**Discussion by JP-Reis** [mathnet/mathnet-numerics#1080] [Q&A] [answered]
Hello community,

I have been trying to figure out how to obtain the regression equation for a linear regression model with a pre-defined intercept value with no success.

The customer want to replicate some of the Excel functionalities that they are used to use in Excel. Is this even possible in with this lib?

Thank you.

**Comment by JP-Reis** [comment]
I know that there is a built-in function to FitThroughOrigin, but what if I want to model the equation to intercept (a=xyz) on Y=3?

**Comment by diluculo** [comment, marked as answer]
You can apply `y - 3.0` to the `FitThroughOrigin` function to model the equation with an intercept of 3 on the y-axis.

**Reply by JP-Reis** [reply]
Ok, so far that seems to do the trick!

Is there any similar function as `FitThroughOrigin ` for exponential and polynomial models?

**Reply by diluculo** [reply]
`var slope = FitThroughOrigin(x, y.Select(p => p - 3.0).ToArray())`

**Reply by diluculo** [reply]
Please see `mathnet-numerics/src/Numerics/Fit.cs`. 

For general models, nonlinear least square fitting (Levenberg-Marquardt) or other optimization methods can be used. Please see the test cases in `mathnet-numerics/src/Numerics.Tests/OptimizationTests`.

**Reply by JP-Reis** [reply]
I'll give it a look. Thank you.
